forastero/bench.py

```diff
@@ -239,9 +241,11 @@ def schedule(self, sequence: BaseSequence, blocking: bool = True) -> None:
         :param blocking: Whether the sequence must complete before the test is
                          allowed to finish
         """
-        task = cocotb.start_soon(sequence(self.fork_log("sequence"), self.random))
+        task = cocotb.start_soon(
+            sequence(self.fork_log("sequence"), self.random, self._arbiter)
+        )
         if blocking:
-            self.sequences.append(task)
+            self._sequences.append(task)
```

Comment:
Could the schedule method return the task? This allows the TB to check when the sequence has completed and to run sequences sequentially i.e.

seq1 = tb.schedule(sequence1())
await seq1
seq2 = tb.schedule(sequence2())
await seq2

Or is there a better/different way to do this?

Reply:
This is a good suggestion will add